src/jit/ssabuilder.cpp

```diff
@@ -973,97 +942,74 @@ void SsaBuilder::TreeRenameVariables(GenTree* tree, BasicBlock* block, SsaRename
         }
     }
 
-    Compiler::IndirectAssignmentAnnotation* pIndirAssign = nullptr;
-    if (!tree->IsLocal() && !IsIndirectAssign(tree, &pIndirAssign))
+    if (!tree->IsLocal())
     {
         return;
     }
 
-    if (pIndirAssign != nullptr)
+    unsigned lclNum = tree->gtLclVarCommon.gtLclNum;
+    // Is this a variable we exclude from SSA?
+    if (m_pCompiler->fgExcludeFromSsa(lclNum))
     {
-        unsigned lclNum = pIndirAssign->m_lclNum;
-        // Is this a variable we exclude from SSA?
-        if (m_pCompiler->fgExcludeFromSsa(lclNum))
+        tree->gtLclVarCommon.SetSsaNum(SsaConfig::RESERVED_SSA_NUM);
+        return;
+    }
+
+    if (tree->gtFlags & GTF_VAR_DEF)
+    {
+        if (tree->gtFlags & GTF_VAR_USEASG)
         {
-            pIndirAssign->m_defSsaNum = SsaConfig::RESERVED_SSA_NUM;
-            return;
+            // This the "x" in something like "x op= y"; it is both a use (first), then a def.
+            // The def will define a new SSA name, and record that in "x".  If we need the SSA
+            // name of the use, we record it in a map reserved for that purpose.
+            unsigned count = pRenameState->CountForUse(lclNum);
+            tree->gtLclVarCommon.SetSsaNum(count);
         }
-        // Otherwise...
-        if (!pIndirAssign->m_isEntire)
```

Comment:
How are non-entire assignments being handled now.
There can be a case like this
```
int  val  = 0x01020304;
(byte*)&val = 0xff;  // val becomes 0x010203ff
```

I believe that we treated such an assignment just like ` val += k;`

Reply:
@briansull - partial defs (non-entire assignments) are already being marked with `GT_VAR_USEASG`; this is how the rest of the JIT identifies these partial defs, and the code in this method (below) is already set up to deal with that case.

Reply:
Good question. That code turns into this IR:
```
     (  6, 10) [000011] ------------              *  STMT      void  (IL 0x006...0x00E)
N001 (  1,  4) [000008] ------------              |  /--*  CNS_INT   int    255
N003 (  6, 10) [000010] -A--G---R---              \--*  ASG       byte  
N002 (  4,  5) [000005] U---G--N----                 \--*  LCL_FLD   byte   V00 loc0         [+0]
```
So it would be modeled using partial defs and that uses a similar hash table - `m_opAsgnVarDefSsaNums`.

Except that doesn't actually in this case because the `val` variable is still marked as address taken and SSA ends up ignoring it (note the above IR dump doesn't have SSA numbers despite being taken post `fgSsaBuild`).